### PR TITLE
Add std:: to some usual suspects

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:** We have several occurrences of missing std namespace.

**Description of the Change:** Add std namespace to usual suspects.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-71935]

### CODE REVIEW

**Review by vincentmr — APPROVED** [review]
LGTM, thanks @AmintorDusko . Don't forget to update the changelog.

File: doc/lightning_qubit/development/add_gate_kernel.rst [hunk 1]
```diff
@@ -21,11 +21,11 @@ In this case, you may first create a file and add a class:
...
            using std::alignment_of_v is sufficient. */
         template <typename PrecisionT>
-        constexpr static size_t required_alignment = std::alignment_of_v<PrecisionT>;
+        constexpr static std::size_t required_alignment = std::alignment_of_v<PrecisionT>;
 
         template <class PrecisionT>
         static void applyPauliX(std::complex<PrecisionT>* data,
-                                size_t num_qubits,
+                                std::size_t num_qubits,
                                 const std::vector<size_t>& wires,
```
Comment (LuisAlfredoNu) [inline doc/lightning_qubit/development/add_gate_kernel.rst:29]:
```suggestion
                                const std::vector<std::size_t>& wires,
```
A missing one

File: doc/lightning_qubit/development/avx_kernels/implementation.rst [hunk 2]
```diff
@@ -42,7 +42,7 @@ The following simple (C++ style) pseudocode shows how the algorithm is implement
 
 .. code-block::
 
-   template<typename PrecisionT, size_t packed_size>
+   template<typename PrecisionT, std::size_t packed_size>
    class ApplyPauliX {
       template<size_t wire>
```
Comment (LuisAlfredoNu) [inline doc/lightning_qubit/development/avx_kernels/implementation.rst:47]:
```suggestion
      template<std::size_t wire>
```
Probably another missing one

**Review by LuisAlfredoNu — COMMENTED** [review]
LGTM just a couple of comments :+1: 
Thank you @AmintorDusko

File: doc/lightning_qubit/development/add_gate_kernel.rst (hunk 1 above)
Comment (AmintorDusko) [reply doc/lightning_qubit/development/add_gate_kernel.rst:29]: Thanks!

File: doc/lightning_qubit/development/avx_kernels/implementation.rst (hunk 2 above)
Comment (AmintorDusko) [reply doc/lightning_qubit/development/avx_kernels/implementation.rst:47]: Thanks!

**Review by multiphaseCFD — APPROVED** [review]
Nice work @AmintorDusko ! Thank you!

**Pull request merged**